R/derive_last_dose.R

```diff
@@ -83,143 +94,147 @@
 #'     head(ex_single, 100),
 #'     filter_ex = (EXDOSE > 0 | (EXDOSE == 0 & grepl("PLACEBO", EXTRT))) &
 #'       nchar(EXENDTC) >= 10,
-#'     dose_start = EXSTDTC,
-#'     dose_end = EXENDTC,
+#'     ex_keep_vars = vars(EXDOSE, EXTRT, EXSEQ, EXENDTC, VISIT),
+#'     dose_date = EXENDTC,
 #'     analysis_date = AESTDTC,
+#'     single_dose_condition = (EXSTDTC == EXENDTC),
 #'     dataset_seq_var = AESEQ,
-#'     new_var = LDOSEDTM,
-#'     output_datetime = TRUE,
-#'     check_dates_only = FALSE,
-#'     traceability_vars = dplyr::vars(LDOSEDOM = "EX", LDOSESEQ = EXSEQ, LDOSEVAR = "EXSTDTC")
+#'     traceability_vars = dplyr::vars(LDOSEDOM = "EX", LDOSESEQ = EXSEQ, LDOSEVAR = "EXENDTC")
 #'   ) %>%
-#'   select(STUDYID, USUBJID, AESEQ, AESTDTC, LDOSEDTM, LDOSEDOM, LDOSESEQ, LDOSEVAR)
+#'   select(STUDYID, USUBJID, AESEQ, AESTDTC, EXDOSE, EXTRT, EXENDTC, LDOSEDOM, LDOSESEQ, LDOSEVAR)
 #'
 derive_last_dose <- function(dataset,
                              dataset_ex,
                              filter_ex = NULL,
                              by_vars = vars(STUDYID, USUBJID),
-                             dose_start,
-                             dose_end,
+                             dose_id = vars(),
+                             dose_date,
                              analysis_date,
                              dataset_seq_var,
```

Comment:
If we call `derive_obs_number` we will still need to use other variables like the grouping or sort order variables. Unless we just use a temporary sequence variable based on `USUBJID` in the `dataset`?

Reply:
Yes, we just need a temporary sequence variable to ensure that after the `filter_extreme()` call the number of observations is the same as in the input dataset. It doesn't matter how the sequence variable is derived as long as it is a unique key of the input dataset. If we number the observations of the input dataset (regardless of `USUBJID`), we could use `by_vars = vars(temp_seq_var)` instead of `by_vars = quo_c(by_vars, temp_seq_var)` in the `filter_extreme()` call.

Reply:
Okay great I've removed the `dataset_seq_var` and replae it with a temporary variable. There was a duplicate warning message that appeared when I used `by_vars = vars(temp_seq_var)` so I've left it as is for now.

Reply:
Where did you get the duplicate warning?

Reply:
Using `by_vars = vars(tmp_seq_var)` a warning message is printed:
Dataset contains duplicate records with respect to `tmp_seq_var` and `tmp_dose_date`
Run `get_duplicates_dataset()` to access the duplicate records

Reply:
If you create the temporary variable by calling
```
    derive_obs_number(new_var = tmp_seq_var)
```
the warning should vanish.

Reply:
Updated